### PR TITLE
Update after selection in Color Picker

### PR DESCRIPTION
**Pull request by hernanmd**
Update each settings browser button color when a color is accepter in the Color Picker window.

Remove unnecessary initializePresenters in StSettingSectionPresenter.

### CODE REVIEW

**Comment by hernanmd** [comment]
Fixes #767

File: src/NewTools-SettingsBrowser/Color.extension.st [hunk 1]
```diff
@@ -3,8 +3,19 @@ Extension { #name : 'Color' }
...
+	| settingPresenter colorValue |
+	
+	colorValue := aSettingDeclaration targetSelector 
+		ifNil: [ aSettingDeclaration realTarget perform: aSettingDeclaration name ]
+		ifNotNil: [ 
+			| settingsTarget |
+			settingsTarget := aSettingDeclaration realTarget perform: aSettingDeclaration targetSelector.
+			settingsTarget perform: aSettingDeclaration getSelector ].
+
+	settingPresenter := (SpPresenter new instantiate: StSettingSHStyleButtonColorPresenterItem on: aSettingDeclaration).
```
Comment (estebanlm) [inline src/NewTools-SettingsBrowser/Color.extension.st:15]: this is equivallent to just `StSettingSHStyleButtonColorPresenterItem on: aSettingDeclaration`, no need of the bureaucracy... and in any case is not recommended to do it like that. Which is the application that presenter belongs to?
(this still can work if you will after add that presenter to another that knows its app, or set the app, but *as is*, it doesn't) 
The recommended way of doing this is by passing a builder (see `SpPresenterBuilder`)

File: src/NewTools-SettingsBrowser/SettingDeclaration.extension.st [hunk 2]
```diff
@@ -10,21 +10,23 @@ SettingDeclaration >> asSettingPresenter [
...
-SettingDeclaration >> getColor [
+SettingDeclaration >> getColorFor: aPresenter [
 
-	(SpPresenter new instantiate: SpColorPickerWindow on: self)
-		whenColorChangedDo: [ : c | 
+	| colorPickerPresenter |
+
+	colorPickerPresenter := aPresenter instantiate: SpColorPickerWindow on: self.
+	colorPickerPresenter
+		whenChangedDo: [ : c |
```
Comment (estebanlm) [inline src/NewTools-SettingsBrowser/SettingDeclaration.extension.st:19]: I prefer real names "aColor" or something like that.

**Review by estebanlm — CHANGES_REQUESTED** [review]
please review those small issues (specially the first one).
notice that *it is possible* that you do not want a builder, in that case, please ensure in the other side (the one that will use the result of `asSettingPresenter`), there is an `owner` set to the resulting presenter.

**Comment by estebanlm** [comment]
and now you need to give me a new PR fixing that, as @Ducasse merged :P

File: src/NewTools-SettingsBrowser/Color.extension.st (hunk 1 above)
Comment (hernanmd) [reply src/NewTools-SettingsBrowser/Color.extension.st:15]: Thanks. I'll address the suggested changes in a new issue

File: src/NewTools-SettingsBrowser/SettingDeclaration.extension.st (hunk 2 above)
Comment (hernanmd) [reply src/NewTools-SettingsBrowser/SettingDeclaration.extension.st:19]: Same as above, I will address the changes in a new issue.

**Pull request merged**